src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
> Such a substitution would not be programmatically sound...if x is now a new custom Real variable that is no longer ordered then you would violate the interface of a Real number.

This is a good point, but is actually in favor of `ForwardDiffNumber <: Real` - `ForwardDiffNumber`s *are* ordered. Now, I'm not sure whether our definition of ordering has a solid basis algebraically, but such a definition is computationally useful for differentiating programs with branches. 

I feel like the most compelling reason to not define `ForwardDiffNumber <: Real` is that such a definition might confuse a user. However, this point is easily countered by the fact `ForwardDiffNumber` types aren't exposed to the user, such that the average user would never need to know they even exist.